### PR TITLE
update MRTextEntity to trigger update for `innerText` manual changes

### PR DESCRIPTION
**Pull request by hanbollar**
## Linking

The actual fix is related to https://discord.com/channels/1137890872688660500/1219366027545149462/1225566696086044693

Most of the cleanup implementation in the pr is related to #546 

## Problem

Updating `innerText` directly only updates the DOM and not the scene itself until there is an `eventUpdate` or direct update call to the 

## Solution

~~initial thought - add it to the main update loop~~

~~NOTE: i'm probably going to swap that in this pr once i finish writing this to a direct - if textContent was modified, do an event trigger update, because checking every loop is hella costly~~

updating directly from the entity trigger √

## Notes

once this pr is in, To see this live - check out the text example: https://examples.mrjs.io/examples/text.html

when locally testing (or before is in): https://localhost:8080/examples/text.html

------------

## Required to Merge

- [x] **PASS** - all necessary actions must pass (excluding the auto-skipped ones)
- [x] **TEST IN HEADSET** - [main dev-testing-example](https://github.com/Volumetrics-io/mrjs/tree/main/samples/index.html) and any of the other [examples](https://github.com/Volumetrics-io/mrjs/tree/main/samples/examples) still work as expected
~~- [ ] **VIDEO** - if this pr changes something visually - post a video here of it in headset-MR and/or on desktop (depending on what it affects) for the reviewer to reference.~~
- [x] **TITLE** - make sure the pr's title is updated appropriately as it will be used to name the commit on merge
~~- [ ] **BREAKING CHANGE**~~
  - **DOCUMENTATION**: This includes any changes to html tags and their components
    - make a pr in the [documentation repo](https://github.com/Volumetrics-io/documentation) that updates the manual docs to match the breaking change
    - link the pr of the documentation repo here: *#pr*
    - that pr must be approved by `@lobau`
  - **SAMPLES/INDEX.HTML**: This includes any changes (html tags or otherwise) that must be done to our landing page submodule as an effect of this pr's updates
    - make a pr in the [mrjs landing page repo](https://github.com/Volumetrics-io/mrjs-landing) that updates the landing page to match the breaking change
    - link the pr of the landing page repo here: *#pr*
    - that pr must be approved by `@hanbollar`

### CODE REVIEW

File: src/core/entities/MRTextEntity.js [hunk 1]
```diff
@@ -40,10 +40,10 @@ export class MRTextEntity extends MRDivEntity {
 
     /**
      * @function
-     * @description (async) sets up the textObject of the text item.
+     * @description Helper for anytime text is manually touched - in connected and by use of `innerText`.
+     * Handles setting `innerText` like expected, sanitizing it for our use, and also making sure the updates are called.
      */
-    async connected() {
-        await super.connected();
+    _textWasManuallyUpdated() {
```
Comment (michaelthatsit) [inline src/core/entities/MRTextEntity.js:46]: Why is this needed? Aren't we checking `entity.textContent == entity.textObj.text` every frame?

Comment (hanbollar) [reply src/core/entities/MRTextEntity.js:46]: nope, we had removed that back when text updating was a big problem - right now we're only checking it when an overall [`eventUpdate` is triggered](https://github.com/Volumetrics-io/mrjs/blob/main/src/core/componentSystems/TextSystem.js#L105) - the `TextSystem`'s update call is empty since it was too costly per frame: https://github.com/Volumetrics-io/mrjs/blob/main/src/core/componentSystems/TextSystem.js#L137

also in the [incoming pr](https://github.com/Volumetrics-io/mrjs/pull/546) later where i rework some of the text system, i bring more of it into the main changes, but i still keep that update empty - didnt want to add all the system changes to this pr as even though it introduces readability and some fixes related to this - a lot of them arent needed until it gets merged

Comment (michaelthatsit) [reply src/core/entities/MRTextEntity.js:46]: hmmm I still think there might be a need for some stuff in the main update. particularly dynamic CSS updates, which we can't detect changes to via events. 

This is good for now but dynamic CSS changes is a table stakes requirement for V1.

Comment (hanbollar) [reply src/core/entities/MRTextEntity.js:46]: s.g, dynamic CSS changes we can add to the main update in https://github.com/Volumetrics-io/mrjs/pull/546 since that's just the add of one function call - i'll keep this pr as is

Comment (hanbollar) [reply src/core/entities/MRTextEntity.js:46]: new issue #562
